contracts/StakingContract.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+import "./Token.sol";
+
+contract StakingContract {
+  
+  Token private token;
+
+  mapping(address => uint256) public etherBalanceOf;
+  mapping(address => uint256) public depositStart;
+  mapping(address => bool) public isDeposited;
```

Comment:
instead of having multiple mapping to store this information, merge these into one mapping to a struct and add the fields you need in that struct